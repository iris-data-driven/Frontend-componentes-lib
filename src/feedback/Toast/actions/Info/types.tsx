export type IconType = {
  type: 'success' | 'error' | 'warning';
};

export type InfoProps = IconType & {
  title: string;
  description: string;
};
