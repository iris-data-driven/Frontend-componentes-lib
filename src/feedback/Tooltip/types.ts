export type TooltipProps = {
  id: string;
  text: string;
  type?: 'small' | 'default'
};
