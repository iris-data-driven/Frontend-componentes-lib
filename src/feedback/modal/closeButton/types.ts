export type CloseButtonType = {
  onClick?: () => void;
};
