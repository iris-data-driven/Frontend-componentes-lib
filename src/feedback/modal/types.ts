import { Styles } from 'react-modal';

type Size = 'medium' | 'large';

export type ModalProps = {
  isOpen: boolean;
  onCloseModal: () => void;
  onAfterClose?: () => void;
  onAfterOpen?: () => void;
  size?: Size;
  contentLabel?: string;
  notClosable?: boolean;
};

export type ModalStyle = (size: Size) => Styles;
