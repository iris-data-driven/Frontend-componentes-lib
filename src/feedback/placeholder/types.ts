type placeholderTypes = 'heading' | 'text' | 'circle' | 'custom';

export type PlaceholderStyleProps = {
  type?: placeholderTypes;
};

export type PlaceholderProps = PlaceholderStyleProps & {
  className?: string;
};
