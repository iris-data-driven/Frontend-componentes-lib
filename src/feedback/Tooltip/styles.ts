import styled from 'styled-components';
import ReactTooltipOriginal from 'react-tooltip';

import Paragraph from 'typography/Paragraph';

import { colors, texts } from 'styles';

export const Span = styled(Paragraph).attrs(() => ({
  as: 'span',
}))`
  ${props => texts.paragraphs[props.type]}
`;

export const ReactTooltip = styled(ReactTooltipOriginal).attrs(props => ({
  place: props['data-type'] === 'small' ? 'top' : 'right',
  effect: 'solid',
  id: props.id,
  border: true,
  textColor: colors.neutral.black,
  backgroundColor: '#f3f4fb',
  borderColor: 'transparent',
  arrowColor: '#f3f4fb',
}))`
  && {
    min-width: ${props => (props['data-type'] === 'small' ? '44px' : '125px')};
    padding: ${props => (props['data-type'] === 'small' ? '0 10px' : '10px')};
    box-sizing: border-box;
    border-radius: 5px;
  }
`;
