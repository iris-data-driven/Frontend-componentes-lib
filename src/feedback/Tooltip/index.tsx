import React, { FC } from 'react';

import { Span, ReactTooltip } from './styles';

import { TooltipProps } from './types';

/**
 * For the tooltip to work, the target element must have the properties 'data-tip' and 'data-for={id}'
 */

const Tooltip: FC<TooltipProps> = ({ id, text, type }) => (
  <>
    <ReactTooltip id={id} data-type={type}>
      <Span type="p3">{text}</Span>
    </ReactTooltip>
  </>
);

Tooltip.defaultProps = {
  type: 'default',
};

export default Tooltip;
