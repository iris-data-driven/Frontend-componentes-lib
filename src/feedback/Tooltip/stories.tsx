import React from 'react';

import Paragraph from 'typography/Paragraph';
import Warning from 'icons/Warning';

import { StoryWrapper } from 'styles/helpers';

import Tooltip from './';

export default {
  title: 'Feedback|Tooltip',
};

export const Small = () => (
  <StoryWrapper>
    <Paragraph type="p3" data-tip data-for="storie-tooltip">
      tooltip
    </Paragraph>
    <Tooltip id="storie-tooltip" text="123" type="small" />
  </StoryWrapper>
);

export const Big = () => (
  <StoryWrapper>
    <Warning data-tip data-for="storie-tooltip" />
    <Tooltip
      id="storie-tooltip"
      text="Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos"
    />
  </StoryWrapper>
);
