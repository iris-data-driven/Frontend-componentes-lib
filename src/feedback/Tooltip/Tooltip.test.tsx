import React from 'react';
import { render, screen } from '@testing-library/react';

import Tooltip from './';

import { TooltipProps } from './types';

describe('Tooltip', () => {
  let props: TooltipProps;
  beforeEach(() => {
    props = {
      id: 'test-tooltip',
      text: '123',
    };
  });

  it('Should render default tooltip correctly', () => {
    props.text = 'Informação adicional';

    render(<Tooltip {...props} />);
    const component = screen.getByText(props.text);

    expect(component).toHaveTextContent('Informação adicional');
    expect(component).toMatchSnapshot();
  });

  it('Should render small tooltip correctly', () => {
    props.type = 'small';

    render(<Tooltip {...props} />);
    const component = screen.getByText(props.text);

    expect(component).toHaveTextContent('123');
    expect(component).toMatchSnapshot();
  });
});
