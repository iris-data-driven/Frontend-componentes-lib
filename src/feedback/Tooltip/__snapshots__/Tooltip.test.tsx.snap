// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Should render default tooltip correctly 1`] = `
<span
  class="sc-AxirZ jFbosC"
  type="p3"
>
  Informação adicional
</span>
`;

exports[`Tooltip Should render small tooltip correctly 1`] = `
<span
  class="sc-AxirZ jFbosC"
  type="p3"
>
  123
</span>
`;
